### PR TITLE
Graceful later_fd fallback if later_api.h is newer than installed later

### LINKED ISSUE #203: Error: function 'execLaterFdNative' not provided by package 'later'
**NOTICE:** If you're seeing this error, `install.packages("later")` and restarting R should fix it. If not, please leave a comment including `packageVersion("later")`

---

This error is happening when a copy of a `LinkingTo: later` package (like httpuv or promises) is installed that was built against later 1.4.0, but the version of later that's actually installed is 1.3.1 or earlier.

This is not academic; it happens for many deployed Shiny apps if they are installing binary packages from CRAN or Posit Package Manager.

For example:

1. Last week you took a snapshot of your app to create a [manifest.json](https://docs.posit.co/connect-cloud/how-to/r/dependencies.html) which contains package names and version numbers. Among the entries are `httpuv 1.6.15` and `later 1.3.1`. All is well.
2. Now you've updated your app logic so you redeploy. The container or whatever is rebuilt. The manifest.json hasn't changed, so httpuv 1.6.15 and later 1.3.1 are installed. But now, httpuv 1.6.15 has been rebuilt against the newest later.
3. When httpuv goes to load, you get the above error; the `inst/include/later.h` is trying to load an RCCallable that doesn't exist.

### PR DESCRIPTION
**Pull request by jcheng5**
Fixes #203 

To test:

```r
devtools::install_github("r-lib/later#204")
install.packages("promises", type="source")
devtools::install_version("later", "1.3.1")
```

then restart R and `library(promises)`

- [x] Un-break inst/include/later.h

### CODE REVIEW

File: inst/include/later.h [hunk 1]
```diff
@@ -48,10 +48,17 @@ namespace later {
...
 // if (LATER_H_API_VERSION != (*dll_api_version)()) { ... }
-#define LATER_H_API_VERSION 2
+#define LATER_H_API_VERSION 3
 #define GLOBAL_LOOP 0
 
 
+// Gets the version of the later API that's provided by the _actually installed_
+// version of later.
+static int apiVersionRuntime() {
+  int (*dll_api_version)() = (int (*)()) R_GetCCallable("later", "apiVersion");
```
Comment (shikokuchuo) [inline inst/include/later.h:58]:
```suggestion
  int (*dll_api_version)(void) = (int (*)(void)) R_GetCCallable("later", "apiVersion");
```

Comment (shikokuchuo) [reply inst/include/later.h:58]: Better to be safe I think...

File: inst/include/later.h [hunk 2]
```diff
@@ -96,6 +103,10 @@ inline void later(void (*func)(void*), void* data, double secs) {
   later(func, data, secs, GLOBAL_LOOP);
 }
 
+static void later_fd_version_error(void (*func)(int *, void *), void *data, int num_fds, struct pollfd *fds, double secs, int loop_id) {
+  Rf_error("later_fd called, but installed version of the 'later' package is too old; please upgrade 'later' to 1.4.1 or above");
```
Comment (shikokuchuo) [inline inst/include/later.h:107]:
```suggestion
  (void) func; (void) data; (void) num_fds; (void) fds; (void) secs; (void) loop_id;
  Rf_error("later_fd called, but installed version of the 'later' package is too old; please upgrade 'later' to 1.4.1 or above");
```

Comment (shikokuchuo) [reply inst/include/later.h:107]: Again, I do this in my functions, just to be safe against -Wunused-variable.

Comment (jcheng5) [reply inst/include/later.h:107]: Guess who grew up on Java, not C 😆

**Pull request merged**